Microsoft.Toolkit.Mvvm/Messaging/Internals/System/Gen2GcCallback.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    /// <summary>
+    /// Schedules a callback roughly every gen 2 GC (you may see a Gen 0 an Gen 1 but only once).
+    /// Ported from https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Gen2GcCallback.cs.
```

Comment:
wondering if we want to make this a permalink or pin to a certain release for historical reference/comparison in the future?

Otherwise since it's just .NET Foundation code as well, we should be fine with just a link from a license perspective. Though wonder if we want a separate thing to call out other .NET Foundation code we've been using somewhere... 🤔

@clairernovotny do other projects do this somehow that we can refer to? Since we're all the same organization under the same license I don't think there's a legal attribution we need to follow, but was just thinking it could be nice from an evangelism perspective.